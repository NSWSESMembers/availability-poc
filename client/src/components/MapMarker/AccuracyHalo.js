import React from 'react';
import PropTypes from 'prop-types';
import { Circle } from 'react-native-maps';

const AccuracyHalo = ({ myPosition }) => {
  if (
    !myPosition ||
    !myPosition.latitude ||
    !myPosition.longitude ||
    !myPosition.accuracy
  ) return null;


  return (
    <Circle
      key="myPositionHalo"
      center={myPosition}
      radius={myPosition.accuracy}
      strokeColor="transparent"
      fillColor="rgba(135,206,250,0.3)"
    />
  );
};

AccuracyHalo.propTypes = {
  myPosition: PropTypes.shape({
    latitude: PropTypes.number,
    longitude: PropTypes.number,
    heading: PropTypes.number,
    accuracy: PropTypes.number,
  }),
};

export default AccuracyHalo;
