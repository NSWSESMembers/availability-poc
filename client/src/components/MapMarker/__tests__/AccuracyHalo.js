import React from 'react';
import renderer from 'react-test-renderer';

import AccuracyHalo from '../AccuracyHalo';

const valid = {
  latitude: 1,
  longitude: 1,
  accuracy: 1,
};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <AccuracyHalo
      myPosition={valid}
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer.create(
    <AccuracyHalo />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();
});
