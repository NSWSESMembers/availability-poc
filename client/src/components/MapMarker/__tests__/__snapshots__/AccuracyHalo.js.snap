// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<AIRMapCircle
  center={
    Object {
      "accuracy": 1,
      "latitude": 1,
      "longitude": 1,
    }
  }
  fillColor="rgba(135,206,250,0.3)"
  radius={1}
  strokeColor="transparent"
  strokeWidth={1}
/>
`;

exports[`renders correctly 2`] = `null`;
